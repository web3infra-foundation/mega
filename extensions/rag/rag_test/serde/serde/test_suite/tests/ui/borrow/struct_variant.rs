use serde_derive::Deserialize;

#[derive(Deserialize)]
struct Str<'a>(&'a str);

#[derive(Deserialize)]
enum Test<'a> {
    #[serde(borrow)]
    S { s: Str<'a> },
}

fn main() {}
