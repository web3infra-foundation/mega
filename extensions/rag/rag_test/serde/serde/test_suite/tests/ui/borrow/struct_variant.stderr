error: #[serde(borrow)] may only be used on newtype variants
 --> tests/ui/borrow/struct_variant.rs:8:5
  |
8 | /     #[serde(borrow)]
9 | |     S { s: Str<'a> },
  | |____________________^
